WPMedia/fusion-news-theme-blocks
# TMEDIA-108 - Convert chains to using Vertical Spacing Component

## Description

Created a new component "VSpace" to provide us the ability to control the vertical spacing of elements, and also have the ability to provide those options to the editor via PageBuilder custom fields

## Jira Ticket
- [TMEDIA-108](https://arcpublishing.atlassian.net/browse/TMEDIA-108)

## Acceptance Criteria
* Add a Custom Field to these chains to allow customers to configure the spacing between blocks: 
    * Single Chain
    * Double Chain
    * Triple Chain
    * Quad Chain
* The custom field should be called Vertical spacing between blocks, it should be a select custom field type, with the following options: 
    * Small
    * Medium
* It should have info text that reads: Select the spacing token used vertically between blocks in this chain. The default in Themes is Medium, except for when placing small promos within a chain, when Small should be selected.  
* The default should be medium, both for new chains added to the page, and for existing chains already placed on pages (so that this is not a breaking change). The Medium choice corresponds to the medium spacing token, which is 24px. If they change this Custom Field to the Small choice, the vertical spacing between blocks should instead be the small spacing token (16px) between blocks. 

## Test Steps

Requires news theme css to be linked locally with PR - https://github.com/WPMedia/news-theme-css/pull/54

1. Checkout this branch `git checkout TMEDIA-108-vertical-spacing-component`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/shared-styles,@wpmedia/single-chain-block,@wpmedia/double-chain-block,@wpmedia/triple-chain-block,@wpmedia/quad-chain-block`
3. Verify that each chain has three new options available to it for configuring the vertical spacing of blocks within a chain
4. Add promo items to each block and adjust the values within the custom fields to verify changes in spacing as desired

## Effect Of Changes
### After

Custom Field controls
<img width="331" alt="VSpace" src="https://user-images.githubusercontent.com/868127/109678223-5192de80-7b72-11eb-9740-f97ee4fd638b.png">

## Dependencies or Side Effects

Requires news theme css to be linked locally with PR - https://github.com/WPMedia/news-theme-css/pull/54

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist 
_The reviewer of the PR should copy-paste this template into the review comments on review._
